dask/layers.py

```diff
@@ -64,43 +74,23 @@ def __dask_distributed_unpack__(cls, state):
         return cls(**state)
 
 
-class BlockwiseCreateArray(Blockwise):
-    """
-    Specialized Blockwise Layer for array creation routines.
+class ArrayChunkShapeDep(ArrayBlockwiseDep):
+    """Produce chunk shapes given a chunk index"""
 
-    Enables HighLevelGraph optimizations.
+    def __getitem__(self, idx: tuple):
+        return tuple(chunk[i] for i, chunk in zip(idx, self.chunks))
 
-    Parameters
-    ----------
-    name: string
-        The output name.
-    func : callable
-        Function to apply to populate individual blocks. This function should take
-        an iterable containing the dimensions of the given block.
-    shape: iterable
-        Iterable containing the overall shape of the array.
-    chunks: iterable
-        Iterable containing the chunk sizes along each dimension of array.
-    """
 
-    def __init__(
-        self,
-        name,
-        func,
-        shape,
-        chunks,
-    ):
-        # Define "blockwise" graph
-        dsk = {name: (func, blockwise_token(0))}
+class ArraySliceDep(ArrayBlockwiseDep):
```

Comment:
If we want users to ever use these directly (as `map_blocks` arguments), I think the docstrings are sparse and names slightly clunky. But I'd be happy with addressing that in a future PR as we get patterns more figured out.

Reply:
For what it's worth, the thought process was `Array{camel-case-the-block-info-key}Dep`, and replace `array-location` with `slice`.

Reply:
Yeah, just thinking that you could also use these with DataFrames, which might make the `Array` part feel odd. And I think `Dep` will end up making sense, but I'm not 100% sure that's the terminology we'll want to use when we end up explaining these things.